code/ATMOSPHERICS/components/components_base.dm

```diff
@@ -134,7 +134,7 @@ Pipenet stuff; housekeeping
 		for(var/I = 1; I <= device_type; I++)
 			if(nodes["n[I]"])
 				if(parents["p[I]"] == reference)
-					return list("n" = nodes["n[I]"])
```

Comment:
you have no idea how many other errors I found, that really just hid runtimes by keeping pipenets from properly building, mainly when constructed.

I'm working on fixing it in my own PR, but I just got one last issue that has been getting me, that I COULD fix with a null check, but the old version of the code didn't need it, so this should.

Anywho, watch this space, You will be say "I actually have no idea what brought me to do this" a bit more =P